python/hail/expr/expression.py

```diff
@@ -3442,73 +3443,105 @@ def in_mito(self):
 class IntervalExpression(Expression):
     """Expression of type :class:`.tinterval`.
 
-    >>> interval = hl.literal(hl.Interval.parse('X:1M-2M'))
+    >>> interval = hl.interval(3, 11)
     """
 
-    @typecheck_method(locus=expr_locus)
-    def contains(self, locus):
-        """Tests whether a locus is contained in the interval.
+    @typecheck_method(value=expr_any)
+    def contains(self, value):
+        """Tests whether a value is contained in the interval.
 
         Examples
         --------
         .. doctest::
 
-            >>> hl.eval_expr(interval.contains(hl.locus('X', 3000000)))
-            False
-
-            >>> hl.eval_expr(interval.contains(hl.locus('X', 1500000)))
+            >>> hl.eval_expr(interval.contains(3))
             True
 
+            >>> hl.eval_expr(interval.contains(11))
+            False
+
         Parameters
         ----------
-        locus : :class:`.Locus` or :class:`.LocusExpression`
-            Locus to test for interval membership.
+        value :
+            Object with type matching the interval point type.
+
         Returns
         -------
         :class:`.BooleanExpression`
-            ``True`` if `locus` is contained in the interval, ``False`` otherwise.
+            ``True`` if `value` is contained in the interval, ``False`` otherwise.
         """
-        locus = to_expr(locus)
-        if self.dtype.point_type != locus.dtype:
-            raise TypeError('expected {}, found: {}'.format(self.dtype.point_type, locus.dtype))
-        return self._method("contains", tbool, locus)
+        value = to_expr(value)
+        if self.dtype.point_type != value.dtype:
+            raise TypeError('expected {}, found: {}'.format(self.dtype.point_type, value.dtype))
+        return self._method("contains", tbool, value)
 
     @property
     def end(self):
-        """Returns the end locus.
+        """Returns the end point.
 
         Examples
         --------
         .. doctest::
 
             >>> hl.eval_expr(interval.end)
-            Locus(contig=X, position=2000000, reference_genome=GRCh37)
+            11
 
         Returns
         -------
-        :class:`.LocusExpression`
-            End locus.
+        :class:`.Expression`
         """
-        return self._field("end", tlocus())
+        return self._field("end", self.dtype)
 
     @property
     def start(self):
-        """Returns the start locus.
+        """Returns the start point.
 
         Examples
         --------
         .. doctest::
 
             >>> hl.eval_expr(interval.start)
-            Locus(contig=X, position=1000000, reference_genome=GRCh37)
+            3
 
         Returns
         -------
-        :class:`.LocusExpression`
-            Start locus.
+        :class:`.Expression`
         """
-        return self._field("start", tlocus())
+        return self._field("start", self.dtype)
 
+    @property
+    def include_start(self):
```

Comment:
awkward naming. Maybe `includes_start`?

Reply:
actually, I like left_inclusive and right_inclusive. Thoughts?

Reply:
I copied the scala naming. @catoverdrive do you have thoughts on this?

Reply:
I like start and end instead of left and right, since that's what we're calling the points. Other than that, I don't have particular thoughts. `start_inclusive` sounds nice.

Reply:
How about `includes_start`?

Reply:
I think the mathematical terms are left-open/-closed and right-open/-closed, see: https://en.wikipedia.org/wiki/Interval_(mathematics)

Reply:
Can that be a different PR?